gitversion/entrypoint.sh

```diff
@@ -2,11 +2,46 @@
 set -o errexit   # abort on nonzero exitstatus
 set -o pipefail  # don't hide errors within pipes
 
-GIT=/git
-GEN=/gen
+if [ ! -z "$GIT_PATH" ]; then
+  # using eval to determine if input is a command or a directory
+  d=$(eval "$GIT_PATH" 2> /dev/null)
+  # directory
+  if [ "$?" != "0" ]; then
+    p=${GIT_PATH}
+  # command
+  else
+    echo GIT_PATH is a command.
+    p=${d}
+  fi
+  # create path if not existent
+  [ ! -d "$p" ] &&  echo creating ${p} && mkdir -p ${p}
+  GIT=${p}
+else
+  # fallback
+  GIT="/git"
+fi
+
+if [ ! -z "$GEN_PATH" ]; then
+  # using eval to determine if input is a command or a directory
+  d=$(eval "$GEN_PATH" 2> /dev/null)
+  # directory
+  if [ "$?" != "0" ]; then
+    p=${GEN_PATH}
+  # command
+  else
+    echo GEN_PATH is a command.
+    p=${d}
+  fi
+  # create path if not existent
+  [ ! -d "$p" ] &&  echo creating ${p} && mkdir -p ${p}
+  GEN=${p}
+else
+  # fallback
+  GEN="/gen"
+fi
```

Comment:
If I only read the entrypoint script ... I will not understand, why you did these changes. It feels cumbersome to adapt the container to concourse needs.

Quote: "This is needed for concourse as it handles input and output resources with a different approach. Those paths are not accessible from within the runner."
That means concouse doesn't allow own volume mounts? If so, isn't that a common problem with a common solution then?

Reply:
If you like you can see these changes more generic. 
The container now has the ability to not run on /git and /gen folders but got a generic way to set these working directories wherever you like. true, this is not needed when running locally with dobi where you just can mount to /gen and /git.0

However for the concourse case this does not work at all. You are fixed to the location /tmp/build/<id>/ (where you don't know which <id> will be used). 
So either you adapt the entrypoint script to make symlinks from /tmp/build/<id>/git to /git or create a way of generic setting the GIT_PATH.

Reply:
I can live with the technical solution for now. If we need more adaptions for usage with concourse in future, I would favour to overwrite the entrypoint with an external one.